crates/ruff_server/src/server.rs

```diff
@@ -111,17 +106,20 @@ impl Server {
     }
 
     pub fn run(self) -> crate::Result<()> {
-        let result = event_loop_thread(move || {
+        event_loop_thread(move || {
             Self::event_loop(
-                &self.conn,
+                &self.connection,
                 &self.client_capabilities,
                 self.session,
                 self.worker_threads,
-            )
+            )?;
+            self.connection.join_io_threads()?;
```

Comment:
Nit: Maybe rename to `close` and document that it will wait for all pending IO to complete.

Reply:
That's funny because I originally called this method `close`. I can go back to that 😄

Reply:
Just FYI I don't think this would wait on active I/O, since we disconnect the message channels before joining the thread(s).